### PR TITLE
improvements to public API

### PR DESCRIPTION
**Pull request by mojavelinux**
- convert attributes in JSON format to Opal hash
- change casing of URI to Uri in methods
- remove trailing endlines
- remove unused delegate method
- change return value of getAttributes() to JSON object
- add alias getDocumentTitle to $doctitle method
- add alias getRevisionDate to $revdate method

### CODE REVIEW

**Comment by mojavelinux** [comment]
Let me know if you need me to split this up to discuss.

File: src/asciidoctor-core-api.js [hunk 1]
```diff
@@ -177,15 +184,15 @@ Opal.Asciidoctor.AbstractNode.$$proto.setOption = function (name) {
   return this.$set_option(name);
 };
 
-Opal.Asciidoctor.AbstractNode.$$proto.getIconURI = function (name) {
+Opal.Asciidoctor.AbstractNode.$$proto.getIconUri = function (name) {
```
Comment (Mogztter) [inline src/asciidoctor-core-api.js]: The reasoning was to keep acronym capitalized but I'm fine with this change

File: src/asciidoctor-core-api.js [hunk 2]
```diff
@@ -292,11 +299,11 @@ Opal.Asciidoctor.Document.$$proto.setTitle = function (title) {
   return this['$title='](title);
 };
 
-Opal.Asciidoctor.Document.$$proto.getDoctitle = function (options) {
+Opal.Asciidoctor.Document.$$proto.getDoctitle = Opal.Asciidoctor.Document.$$proto.getDocumentTitle = function (options) {
```
Comment (Mogztter) [inline src/asciidoctor-core-api.js]: :+1:

File: src/asciidoctor-core-api.js [hunk 3]
```diff
@@ -292,11 +299,11 @@ Opal.Asciidoctor.Document.$$proto.setTitle = function (title) {
...
   return this.$doctitle(toHash(options));
 };
 
-Opal.Asciidoctor.Document.$$proto.getRevdate = function () {
+Opal.Asciidoctor.Document.$$proto.getRevdate = Opal.Asciidoctor.Document.$$proto.getRevisionDate = function () {
```
Comment (Mogztter) [inline src/asciidoctor-core-api.js]: :+1:

**Comment by Mogztter** [comment]
PhantomJS does not support `Object.assign`:

```
PhantomJS 2.1.1 (Linux 0.0.0) CommonJS Loading should load document attributes FAILED
	TypeError: undefined is not a function (evaluating 'Object.assign({}, this.attributes.$$smap)') in /tmp/b5c495e0da5e44a8f41e19766210aff6.browserify (line 35926)
	getAttributes@/tmp/b5c495e0da5e44a8f41e19766210aff6.browserify:35926:23 <- build/asciidoctor.js:35924:0
	/tmp/b5c495e0da5e44a8f41e19766210aff6.browserify:80917:33 <- spec/share/common-spec.js:58:0
```

File: src/asciidoctor-core-api.js [hunk 4]
```diff
@@ -292,14 +326,22 @@ Opal.Asciidoctor.Document.$$proto.setTitle = function (title) {
...
   return this.$revdate();
 };
 
+Opal.Asciidoctor.Document.$$proto.getRevnumber = Opal.Asciidoctor.Document.$$proto.getRevisionNumber = function () {
+  return this.getAttributes().revnumber;
```
Comment (Mogztter) [inline src/asciidoctor-core-api.js]: @mojavelinux Do you think Revnumber and Revremark should be in the Asciidoctor core API ?

**Comment by Mogztter** [comment]
Why continuous integration does not run on this pull request :thinking:

**Comment by mojavelinux** [comment]
> Why continuous integration does not run on this pull request

A few days ago the webhooks were broken. I think if I force push it will fire again.

File: src/asciidoctor-core-api.js [hunk 5]
```diff
@@ -5,32 +5,43 @@ var toHash = function (object) {
   return object;
 };
 
+var prepareOptions = function (options) {
```
Comment (mojavelinux) [inline src/asciidoctor-core-api.js]: That is correct. It is now possible to pass the attributes as a Hash. We dig down to grab the value of the attributes key and coerce it if necessary. We're trying to shield the user from the low-level details here.

+1 to a test.

File: src/asciidoctor-core-api.js [hunk 6]
```diff
@@ -67,10 +78,6 @@ Opal.Asciidoctor.AbstractBlock.$$proto.convert = function () {
   return this.$convert();
 };
 
-Opal.Asciidoctor.AbstractBlock.$$proto.delegate = function () {
```
Comment (mojavelinux) [inline src/asciidoctor-core-api.js]: I honestly have no idea what this method refers to. "delegate" isn't anywhere in Asciidoctor core.

File: src/asciidoctor-core-api.js [hunk 7]
```diff
@@ -90,7 +97,7 @@ Opal.Asciidoctor.AbstractBlock.$$proto.convert = function () {
 // AbstractNode API
 
 Opal.Asciidoctor.AbstractNode.$$proto.getAttributes = function () {
-  return this.attributes;
+  return Object.assign({}, this.attributes.$$smap);
```
Comment (mojavelinux) [inline src/asciidoctor-core-api.js]: +1 to test.

File: src/asciidoctor-core-api.js (hunk 1 above)
Comment (mojavelinux) [inline src/asciidoctor-core-api.js]: I'm trying to align with AsciidoctorJ here. See http://static.javadoc.io/org.asciidoctor/asciidoctorj/1.6.0-alpha.3/org/asciidoctor/ast/ContentNode.html

**Comment by mojavelinux** [comment]
> PhantomJS does not support Object.assign

Oops, I didn't realize that. I was just being lazy...we could just do a manual clone like https://gist.github.com/wycliffepeart/c6520bc1c340fd0c59af5b7cb59ea610 (a bit simpler than the polyfill you linked to).

**Comment by oncletom** [comment]
If it's only for testing purposes, you can also consider using https://qa.polyfill.io/v2/docs/
It solved all my testing issues with Phantom.

**Comment by Mogztter** [comment]
Someone could use Phantomjs as a runtime but this is mostly for testing purposes.

**Comment by Mogztter** [comment]
This looks good to me, thanks 😊

File: src/asciidoctor-core-api.js (hunk 4 above)
Comment (Mogztter) [inline src/asciidoctor-core-api.js]: @mojavelinux This proposal had been removed (force pushed), you don't like it ?

**Comment by mojavelinux** [comment]
@Mogztter I was wondering what happened the revnumber change. Somehow, it got lost in the pushes. Let's take it up as a separate issue.

**Pull request merged**